code/core/system/proxy-velodyne16/src/ProxyVelodyne16.cpp

```diff
@@ -55,25 +54,31 @@ void ProxyVelodyne16::setUp() {
     m_udpPort = getKeyValueConfiguration().getValue< uint32_t >("proxy-velodyne16.udpPort");
     m_udpreceiver = UDPFactory::createUDPReceiver(m_udpReceiverIP, m_udpPort);
 
-    m_CompactPointCloud = getKeyValueConfiguration().getValue< bool >("proxy-velodyne16.CompactPointCloud") == 1;
+    m_pointCloudOption = getKeyValueConfiguration().getValue< uint32_t >("proxy-velodyne16.pointCloudOption");
+    std::cout<<"Point cloud option:"<<m_pointCloudOption<<std::endl;
     
-    if(!m_CompactPointCloud) {
+    if(m_pointCloudOption==0 || m_pointCloudOption==2){
         m_memoryName = getKeyValueConfiguration().getValue< string >("proxy-velodyne16.sharedMemory.name");
         m_memorySize = getKeyValueConfiguration().getValue< uint32_t >("proxy-velodyne16.sharedMemory.size");
         m_velodyneSharedMemory = SharedMemoryFactory::createSharedMemory(m_memoryName, m_memorySize);
-
-        m_velodyne16decoder = shared_ptr< Velodyne16Decoder >(new Velodyne16Decoder(m_velodyneSharedMemory, getConference(), getKeyValueConfiguration().getValue< string >("proxy-velodyne16.calibration")));
-
-        m_udpreceiver->setStringListener(m_velodyne16decoder.get());
-        // Start receiving bytes.
-        m_udpreceiver->start();
+        if(m_pointCloudOption==0){
+            m_velodyne16decoder = shared_ptr< Velodyne16Decoder >(new Velodyne16Decoder(m_velodyneSharedMemory, getConference(), getKeyValueConfiguration().getValue< string >("proxy-velodyne16.calibration"),false));
+        }
+        else{
+            m_velodyne16decoder = shared_ptr< Velodyne16Decoder >(new Velodyne16Decoder(m_velodyneSharedMemory, getConference(), getKeyValueConfiguration().getValue< string >("proxy-velodyne16.calibration"),true));
+        }
     }
-    else {
-        m_velodyne16decoderCPC = shared_ptr< Velodyne16DecoderCPC >(new Velodyne16DecoderCPC(getConference()));
-        m_udpreceiver->setStringListener(m_velodyne16decoderCPC.get());
-        // Start receiving bytes.
-        m_udpreceiver->start();
+    else if(m_pointCloudOption==1){
+        m_velodyne16decoder = shared_ptr< Velodyne16Decoder >(new Velodyne16Decoder(getConference()));
     }
+    else{
+        std::cout<<"Invalid point cloud option! 0: shared point cloud (SPC) only; 1: compact point cloud (CPC) only; 2: both SPC and CPC"<<std::endl;
+        return;
```

Comment:
In this case, the velodyne decoder will not be instantiated and there will be no UDP string listener. So no SPC or CPC is sent.